packages/block-library/src/search/style.scss

```
@@ -97,6 +97,15 @@ $button-spacing-y: math.div($grid-unit-15, 2); // 6px
 	}
 }
 
+// We are lowering the specificity so that the input element can override the rule for the input element inside the search block.
+:where(.wp-block-search__input) {
+	font-family: inherit;
+	font-weight: inherit;
+	font-size: inherit;
+	line-height: inherit;
+	letter-spacing: inherit;
```

```suggestion
	letter-spacing: inherit;
	text-transform: inherit;
```

---

```suggestion
	letter-spacing: inherit;
	text-transform: inherit;
	font-style: inherit;
```

I think we're missing one more typography related style. Text decoration and text columns are rightfully omitted.